### PR TITLE
Adapt to type sent by DCS + macros to populate and read the GRP DCS config entry

### PR DESCRIPTION
**Pull request by chiarazampolli**

### CODE REVIEW

**Comment by chiarazampolli** [comment]
Hello @aphecetche , @wiechula , @mpoghos , @syano0822, @shahor02 , @noferini 

This should fix the issue with the fact that ADAPOS Is now sending DPVAL instead of RAW types (@wiechula, @martenole  : you might need to fix this in your ongoing development for DCS DataPoints processing).

I tested the changes in GRP + getValue function of DataPointCompositeObject, and it worked. 

Chiara

**Comment by chiarazampolli** [comment]
Changed to WIP to see how today's test behave.

Chiara

**Comment by chiarazampolli** [comment]
Working with today's tests for MFT, TOF, MCH, EMCAL, GRP, so moving to ready for review.

**Comment by shahor02** [comment]
Hi @chiarazampolli there are conflicts to resolve...

**Comment by chiarazampolli** [comment]
Hi @shahor02 , 

Yes, because meanwhile also detectors committed similar fixes :-) 
I will take care, I need also to move the GRP CCDB object from Calib to Config, as we discussed today. 

Chiara

**Comment by chiarazampolli** [comment]
@shahor02 , conflicts solved. 
I think that this PR is ready, after the tests are passed.

**Comment by chiarazampolli** [comment]
Hi @shahor02 , 
At your convenience, could you check this PR?
Thanks!
Chiara

File: Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx [hunk 1]
```diff
@@ -414,7 +414,7 @@ void GRPDCSDPsProcessor::updateLHCIFInfoCCDB()
   }
   std::map<std::string, std::string> md;
   md["responsible"] = "Chiara Zampolli";
-  o2::calibration::Utils::prepareCCDBobjectInfo(mLHCInfo, mccdbLHCIFInfo, "GLO/Config/LHCIF", md, mStartValidity, o2::calibration::Utils::INFINITE_TIME);
+  o2::calibration::Utils::prepareCCDBobjectInfo(mLHCInfo, mccdbLHCIFInfo, "GLO/Config/LHCIF", md, mStartValidity, mStartValidity + 3 * 3600000); // valid for 3 days
```
Comment (shahor02) [inline Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx]: It is 3 hours, not 3 days, can you change it to mStartValidity + 24L*3*3600000;  ? (Note `L`)
Same comment for the lines below.

**Review by shahor02 — CHANGES_REQUESTED** [review]

File: Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx (hunk 1 above)
Comment (chiarazampolli) [reply Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx]: Yes, sorry, I forgot to do 24*, and copied pasted. I will fix it.
Why 24L33600000? Why twice “3”?

chiara

Comment (chiarazampolli) [reply Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx]: Or did you mean 24L*3600000?

Comment (chiarazampolli) [reply Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx]: Ok it is for the 3 days, now i got it. Sorry, long day…

Comment (shahor02) [reply Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx]: Github has swallowed `*`, I meant `24L*3*3600000`

Comment (chiarazampolli) [reply Detectors/GRP/calibration/src/GRPDCSDPsProcessor.cxx]: Yes, yes, I implemented it like that. I thought it meant something special initially :-) 
Marking as resolved.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**